src/main/java/com/photolocos/enterprise/dao/ILocationDAO.java

```diff
@@ -35,7 +35,7 @@
      * @param latitude  the latitude of the location that the user wants to get more info about photography locations
      * @return the location object matching the co-ordinates, if not found return null
      */
-    LocationDTO fetchByCoordinates(String longitude, String latitude);
+    LocationDTO fetchByCoordinates(double longitude, double latitude) throws Exception;
 
```

Comment:
DAO layer has the potential to throw Exceptions. Also made longitude and latitude a double instead of a String. Consider making them BigDecimal if greater precision is needed.

Reply:
I'd go with BigDecimal here.  Between String and double, I'd go with String.  double has floating point arithmetic issues.  In GPS, the positions to the right of the decimal are significant, so this can throw off GPS locations.  Hence, I stick with String or BigDecimal.